node/src/components/linear_chain_sync.rs

```diff
@@ -291,7 +291,7 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
                     info!(%block_height, "Finished synchronizing linear chain up until trusted hash.");
                     let peer = self.random_peer_unsafe();
                     // Kick off syncing trusted hash descendants.
-                    self.state = State::sync_descendants(trusted_hash);
+                    self.state = State::sync_descendants(trusted_hash, block_header);
```

Comment:
Is this correct? `block_handled` is called when we have just executed a block and handled it in the consensus. Here, it's passed as a header of a block that we _are downloading_ but haven't yet handled and executed. That's why this field is populated as `None` at the start of `SyncingDescendants` state – b/c there's no block that we downloaded but are waiting for an execution.